BDCSCli/Recipe.hs

```diff
@@ -260,8 +260,66 @@ deleteFile repo branch filename = do
 
     -- Create a new commit: repositoryCreateCommit
     Git.repositoryCreateCommit repo ref sig sig encoding message tree [parent_commit]
-    return ()
 
+revertFile :: Git.Repository -> T.Text -> T.Text -> T.Text -> IO Git.OId
+revertFile repo branch filename commit = do
+    mcommit_id <- Git.oIdNewFromString commit
+    -- XXX Handle errors
+    let commit_id = fromJust mcommit_id
+    revertFileCommit repo branch filename commit_id
+
+revertFileCommit :: Git.Repository -> T.Text -> T.Text -> Git.OId -> IO Git.OId
+revertFileCommit repo branch filename commit_id = do
+    mcommit_obj <- Git.repositoryLookupCommit repo commit_id
```

Comment:
This function would be a perfect candidate to use MaybeT to avoid having to do all the fromJust stuff yourself.  I can work on that separately, if you would like.  I see no reason to hold up the PR for a change that doesn't actually affect the function of it.